encoding/length.go

```diff
@@ -0,0 +1,94 @@
+package encoding
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/xichen2020/eventdb/generated/proto/encodingpb"
+	"github.com/xichen2020/eventdb/x/unsafe"
+)
+
+type lengthStringIterator struct {
+	reader    io.Reader
+	meta      encodingpb.StringMeta
+	container []byte
+	curr      string
+	err       error
+	closed    bool
+	zero      uint64
+}
+
+func newLengthStringIterator(reader io.Reader, meta encodingpb.StringMeta) ForwardStringIterator {
+	return &lengthStringIterator{
+		reader:    reader,
+		meta:      meta,
+		container: make([]byte, 8),
+	}
+}
+
+func (d *lengthStringIterator) Next() bool {
+	if d.closed {
+		return false
+	}
+
+	_, d.err = d.reader.Read(d.container[:d.meta.BytesPerRecord])
+	if d.err != nil {
+		return false
+	}
+
+	lengthBytes := int(binary.LittleEndian.Uint64(d.container[:8]))
+
+	d.container = EnsureBufferSize(d.container, lengthBytes)
+	_, d.err = d.reader.Read(d.container[:lengthBytes])
+	if d.err != nil {
+		return false
+	}
+
+	d.curr = string(d.container[:lengthBytes])
```

Comment:
I was doing this on purpose because I was assuming that the `string` would need to be valid across `Next()` calls. On second thought, it will in some cases (e.g. raw data since we need to return aggregate results) and not in other cases (filtering to get doc IDs). 

I'll have it return a `[]byte` which will let the user figure out whether to allocate a `string` or not.

Reply:
Discussed in person, will keep the return value as is and use the `unsafe` method to convert to `string` but make a note that the caller will have to copy the `string` to have a valid reference btwn `Next()` calls.